mesonbuild/utils/universal.py

```diff
@@ -1090,36 +1093,40 @@ def has_path_sep(name: str, sep: str = '/\\') -> bool:
     return False
 
 
+def cmd_quote(arg: str) -> str:
+    # see: https://docs.microsoft.com/en-us/windows/desktop/api/shellapi/nf-shellapi-commandlinetoargvw#remarks
+
+    if arg and not re.search('[ \t\n\r"]', arg):
```

Comment:
I think it's better to keep caching the re.compile() value, this is called a lot AFAIK.

Reply:
Hm, I did not know python cached that already. Probably not worth it then, ok.